src/prism.c

```diff
@@ -6236,8 +6241,13 @@ pm_symbol_node_to_string_node(pm_parser_t *parser, pm_symbol_node_t *node) {
     pm_string_node_t *new_node = PM_ALLOC_NODE(parser, pm_string_node_t);
     pm_node_flags_t flags = 0;
 
-    if (parser->frozen_string_literal) {
-        flags = PM_NODE_FLAG_STATIC_LITERAL | PM_STRING_FLAGS_FROZEN;
+    switch (parser->frozen_string_literal) {
+        case 1:
+            flags = PM_NODE_FLAG_STATIC_LITERAL | PM_STRING_FLAGS_FROZEN;
+            break;
+        case 2:
```

Comment:
```suggestion
        case -1:
```
Right?

Reply:
Yes 🤦

Reply:
Well, `0` but similarly broken.

Also perhaps I don't understand the meaning of `PM_NODE_FLAG_STATIC_LITERAL` correctly, but I think it applies regardless of frozen string literals.

As long as there is no interpolation, I think this flag should be set cause the string can be created eagerly. Frozen string literals will just define whether we have to `dupstring` or `putobject`.